src/Chapter2.hs

```diff
@@ -336,7 +352,13 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList x y l =
+ if length l == 0 || y < x || y < 0 || x < 0 || x > length l
```

Comment:
Calculating list length is an expensive operation. It's always better to avoid using `length` is possible. For example, using the `null` function to check if the list is empty is better than using `length`. Also, you don't need to compare left bound with list length. `drop` will take care of it and it is as efficient as using `length`